src/root/body/index.tsx

```diff
@@ -441,106 +488,107 @@ const Body: React.FC<params> = (props) => {
     if (ctx === null) {
       return;
     }
-    image.onload = () => {
-      canvas.width = image.width;
-      canvas.height = image.height;
-      ctx.drawImage(image, 0, 0);
-      imageCache.forEach((storedImage) => {
-        // find the current image in the image cache based on current index
-        if (storedImage.index === imageIndex && storedImage.annotated) {
-          storedImage.classifications.forEach((prediction, index) => {
-            // !storedImage.overlapping[index]     REMOVE THIS TO SHOW ONLY 1 BB
-            if (
-              storedImage.scores[index] >= scoreThreshold / 100 &&
-              (prediction === selectedLabel || selectedLabel === "all")
-            ) {
-              ctx.beginPath();
-              // draw label index
-              ctx.font = "bold 0.9vw Arial";
-              ctx.fillStyle = "black";
-              ctx.textAlign = "center";
-              Object.keys(labelOccurrences).forEach((key, labelIndex) => {
-                const scorePercentage = (
-                  storedImage.scores[index] * 100
-                ).toFixed(0);
-                // check to see if label is cut off by the canvas edge, if so, move it to the bottom of the bounding box
-                if (storedImage.boxes[index].topY <= 40) {
-                  if (prediction === key) {
-                    if (switchTable) {
-                      ctx.fillText(
-                        `[${labelIndex + 1}] - ${scorePercentage}%`,
-                        ((storedImage.boxes[index].bottomX as number) -
-                          (storedImage.boxes[index].topX as number)) /
-                          2 +
-                          (storedImage.boxes[index].topX as number),
-                        (storedImage.boxes[index].bottomY as number) + 23,
-                      );
-                    } else {
-                      ctx.fillText(
-                        `[${index + 1}]`,
-                        ((storedImage.boxes[index].bottomX as number) -
-                          (storedImage.boxes[index].topX as number)) /
-                          2 +
-                          (storedImage.boxes[index].topX as number),
-                        (storedImage.boxes[index].bottomY as number) + 23,
-                      );
-                    }
-                  }
+
+    // if Tiff image, convert to PNG
+    if (imageSrc.includes("image/tiff")) {
+      await decodeTiff()
+        .then((dTiff: DecodedTiff) => {
+          const { rgba, width, height } = dTiff;
+          imgWidth = width;
+          imgHeight = height;
+          canvas.width = imgWidth;
+          canvas.height = imgHeight;
+          const imgd = ctx.createImageData(imgWidth, imgHeight);
+          for (let i = 0; i < rgba.length; i += 1) {
+            imgd.data[i] = rgba[i];
+          }
+          ctx.putImageData(imgd, 0, 0);
+        })
+        .catch((error) => {
+          if (error instanceof Error) {
```

Comment:
I think @rngadam already mentioned it to you, but throwing generic Errors can result in problems being caught when we would like to know about them.